WordPress/src/test/java/org/wordpress/android/util/SnackbarSequencerConcurrentTest.kt

```diff
@@ -76,42 +81,91 @@ class SnackbarSequencerConcurrentTest {
         // Then
         verify(wpSnackbar, times(1)).show()
         advanceTimeBy(checkPoints[0])
-        verify(wpSnackbar, times(1)).show()
+        verify(wpSnackbar, times(2)).show()
         advanceTimeBy(checkPoints[1])
+        verify(wpSnackbar, times(3)).show()
+        advanceTimeBy(checkPoints[2])
+
+        // Nothing else shown
+        verify(wpSnackbar, times(3)).show()
+        verifyNoMoreInteractions(wpSnackbar)
+    }
+
+    @Test
+    fun `snackbars are not shown until previous duration elapsed`() = runBlockingTest(coroutineContext) {
+        // Given
+        val items = getItems()
+        val checkPoints = getCheckPoints(items)
+
+        // When
+        for (item in items) {
+            sequencer.enqueue(item)
+        }
+
+        // Then
+        verify(wpSnackbar, times(1)).show()
+        advanceTimeBy(checkPoints[0] - 2 * SNACKBAR_DURATION_MARGIN)
```

Comment:
We might want to consider removing the `2 *` here or setting `SNACKBAR_DURATION_MARGIN` to `200L`. Wdyt?

Reply:
Btw I'm also not sure the name makes sense - it's basically just a random number right? We might want to consider hardcoding `100`. I think it's clear enough when used only within 3 lines of code. Wdyt?

Reply:
This pairs a bit with the previous comment. Basically in respect to the check points I want to sample the state 1 SNACKBAR_DURATION_MARGIN before the snackbar duration end (so the 2* accounts for the initial offset in the checkpoints). I agree the naming is not great at all! I wait for your suggestion on keeping the extra margin in the checkpoints above (in that case maybe better to keep also the margin as parametric, otherwise we can get rid of it)

Reply:
So the unnecessary extra margins adds one more level of complexity. This make me even more unsure whether it's worth it. Wdyt @planarvoid ?

Reply:
I think this test is quite complicated and it doesn't have to be. For example why do we need different `TEST_DURATIONS` and not just have the same duration for all the items? And maybe we can just have 2 and not 3. These make it feel not like a unit test but more like something testing more than 1 item.

Saying that, I think we can merge this test and the previous one, have just one duration (500 ms), do something like:
- verify item 1 was emitted
- wait 250 ms
- verify nothing was emitted
- wait 250 ms
- verify item 2 was emitted

Reply:
>I think this test is quite complicated and it doesn't have to be. For example why do we need different TEST_DURATIONS and not just have the same duration for all the items? And maybe we can just have 2 and not 3 items? These make it feel not like a unit test but more like something testing more than 1 item.

Good point! ;)

>Saying that, I think we can merge this test and the previous one, have just one duration (500 ms), do something like:

Actually I suggested splitting the test in two tests. I think it's nice for the test to have only a single reason to fail = test only one behavior. The fact the bodies of the two tests are almost identical isn't that important imo. I kind of like that the name of each of the tests explicitly says what it does. Wdyt?

Reply:
Good points from both of you here 👍! I applied a bit from both 😊.
Simplified the tests complexity reducing to 2 elements and using same duration from all snackbars. Also kept the 2 tests separated since I also feel this makes the test intent more clear. Let me know wdyt 😊. Changes here 353e66d85e25c4cb24cb050ae9228a85ce46655e.

Reply:
Thanks for the changes, it looks much simpler and more readable. I think you're both right and it's correct to keep both tests 👍